### PR TITLE
feat(route): migrate to v2 api (twreporter)

### PR DESCRIPTION
**Pull request by emdoe**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/twreporter/category/world
/twreporter/newest
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/twreporter/fetch-article.ts [hunk 1]
```diff
@@ -1,67 +1,99 @@
...
-            // An article may have multiple authors
-            const name = [];
-            for (const item of $('.cidPTd > a').get()) {
-                const $ = load(item);
-                name.push($('.fJSaZP').text());
-            }
+    const banner_image = post.og_image.resized_targets.desktop.url;
+    const caption = post.leading_image_description;
+    const banner_description = post.og_image.description;
+    const og_description = post.og_description;
```
Comment (TonyRL) [inline lib/routes/twreporter/fetch-article.ts:39]: `camelCase`
https://docs.rsshub.app/joinus/advanced/script-standard

File: lib/routes/twreporter/fetch-article.ts [hunk 2]
```diff
@@ -1,67 +1,99 @@
...
+                    case 'youtube': {
+                        const video = content[0].youtubeId;
+                        const id = video.split('?')[0];
+                        block = art(path.join(__dirname, 'templates/youtube.art'), { video: id });
+
+                        break;
+                    }
+                    default:
+                        block = `${content}<br>`;
+                }
```
Comment (TonyRL) [inline lib/routes/twreporter/fetch-article.ts:85]: Please move the switch statement out of the map loop.

Comment (emdoe) [reply lib/routes/twreporter/fetch-article.ts:85]: Sorry, I'm a little confused, do you mean to use for loop instead? (Note that I have to use `switch(item.type)`.)

Comment (TonyRL) [reply lib/routes/twreporter/fetch-article.ts:85]: > Please move the switch statement out of the map loop.

```diff
+ function switchFn() {}

const text = post.content.api_data
    .map(() => {
        ...
-        switch() {}
+        switchFn()
        ...
});

**Pull request merged**